provider/cmd/pulumi-resource-nuage/nuage_provider/serverless_database.py

```diff
@@ -0,0 +1,338 @@
+        data_api: Optional[pulumi.Input[bool]],
+        s3_extension: Optional[pulumi.Input[bool]],
+    ) -> None:
+        self.resource_name = resource_name
+        self.vpc_id = vpc_id
+        self.vpc_subnets = vpc_subnets
+        self.database_type = database_type
+        self.database_name = database_name
+        self.master_username = master_username
+        self.ip_whitelist = ip_whitelist
+        self.skip_final_snapshot = skip_final_snapshot
+        self.data_api = data_api
+        self.s3_extension = s3_extension
+
+
+class ServerlessDatabase(pulumi.ComponentResource):
+    user: pulumi.Output[str]
+    password: pulumi.Output[str]
+    host: pulumi.Output[str]
+    port: pulumi.Output[int]
+    name: pulumi.Output[str]
+    cluster_arn: pulumi.Output[str]
+    uri: pulumi.Output[str]
+
+    def __init__(
+        self,
+        name: str,
+        args: ServerlessDatabaseArgs,
+        props: Optional[dict] = None,
+        opts: Optional[pulumi.ResourceOptions] = None,
+    ) -> None:
+        super().__init__("nuage:aws:ServerlessDatabase", name, props, opts)
+
+        # RDS subnet group
+        subnet_group = aws.rds.SubnetGroup(
+            resource_name=f"{args.resource_name}-subnet-group",
+            name_prefix=f"{name}-",
+            description=f"{name}",
+            subnet_ids=args.vpc_subnets,
+            tags={"Name": f"{name} subnet group"},
+            opts=pulumi.ResourceOptions(parent=self, replace_on_changes=["subnet_ids"]),
+        )
+
+        # Cluster master password
+        aurora_master_password = pulumi_random.RandomPassword(
+            resource_name=f"{args.resource_name}-master-password",
+            length=30,
+            special=True,
+            # https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Limits.html#RDS_Limits.Constraints
+            override_special=string.punctuation.translate(str.maketrans("", "", '/"@')),
+            opts=pulumi.ResourceOptions(parent=self),
+        )
+
+        security_group = aws.ec2.SecurityGroup(
+            resource_name=f"{args.resource_name}/security-group",
+            name_prefix=f"{name}-database-sg",
+            vpc_id=args.vpc_id,
+            opts=pulumi.ResourceOptions(parent=self),
+        )
+
+        if args.database_type == "mysql":
+            port = 3306
+        else:
+            # PostgreSQL default
+            port = 5432
+
+        aws.ec2.SecurityGroupRule(
+            resource_name=f"{args.resource_name}/security-group/ingress-rule",
+            security_group_id=security_group.id,
+            type="ingress",
+            protocol=aws.ec2.ProtocolType.TCP,
+            from_port=port,
+            to_port=port,
+            cidr_blocks=args.ip_whitelist or ["0.0.0.0/0"],
+            description="Allow inbound connectivity to the database",
+            opts=pulumi.ResourceOptions(parent=security_group),
+        )
+
+        if args.database_type == "mysql":
```

Comment:
I'm not a fan of this big if / else 

Can you refactor to use a single cluster but use if/else in each variable (or an enum with the variations, or anything else)